Launching docker version of the server over https

Hi,
i've been trying along few stuff to get the transfer.sh server working only with https and using the docker image but was not successful at all.
Would you mind adding an example on how to launch it with https support ?
I've naively tried:
`docker run -v /tmp/truc:/tmp --publish 8081:443 dutchcoders/transfer.sh:latest --tls-listener :443 --provider local  --basedir /tmp/
2017/10/16 14:47:24 Transfer.sh server started.
listening on port: :8080
using temp folder: /tmp
using storage provider: local
2017/10/16 14:47:24 ---------------------------
panic: open : no such file or directory

goroutine 143 [running]:
panic(0x833740, 0xc420730030)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/dutchcoders/transfer.sh/server.(*Server).Run.func5(0xc420182750, 0xc4206fd4a0)
	/go/src/github.com/dutchcoders/transfer.sh/server/server.go:353 +0x12c
created by github.com/dutchcoders/transfer.sh/server.(*Server).Run
	/go/src/github.com/dutchcoders/transfer.sh/server/server.go:355 +0x169f`
It seems it does not like when --listener is not given.
Any help appreciate.
Best
NB: Great job by the way ! We are using it for a C programming classroom where we ask for some image manipulation and it helps us by providing a temporary storage for the result images.

--- a/README.md
+++ b/README.md
@@ -53,7 +53,8 @@ listener | port to use for http (:80) | |
 profile-listener | port to use for profiler (:6060)| |
 force-https | redirect to https | false |
 tls-listener | port to use for https (:443) | |
-tls-cert-file | path to tls certificate | | 
+tls-listener-only | flag to enable tls listener only | |
+tls-cert-file | path to tls certificate | |
 tls-private-key | path to tls private key | |
 http-auth-user | user for basic http auth on upload | |
 http-auth-pass | pass for basic http auth on upload | |

--- a/cmd/cmd.go
+++ b/cmd/cmd.go
@@ -54,6 +54,10 @@ var globalFlags = []cli.Flag{
 		Usage: "127.0.0.1:8443",
 		Value: "",
 	},
+	cli.BoolFlag{
+		Name:  "tls-listener-only",
+		Usage: "",
+	},
 	cli.StringFlag{
 		Name:  "tls-cert-file",
 		Value: "",
@@ -204,8 +208,11 @@ func New() *Cmd {
 			options = append(options, server.Listener(v))
 		}
 
-		if v := c.String("tls-listener"); v != "" {
-			options = append(options, server.TLSListener(v))
+		if v := c.String("tls-listener"); v == "" {
+		} else if c.Bool("tls-listener-only") {
+			options = append(options, server.TLSListener(v, true))
+		} else {
+			options = append(options, server.TLSListener(v, false))
 		}
 
 		if v := c.String("profile-listener"); v != "" {

--- a/server/server.go
+++ b/server/server.go
@@ -95,9 +95,10 @@ func UserVoice(userVoiceKey string) OptionFn {
 	}
 }
 
-func TLSListener(s string) OptionFn {
+func TLSListener(s string, t bool) OptionFn {
 	return func(srvr *Server) {
 		srvr.TLSListenerString = s
+		srvr.TLSListenerOnly = t
 	}
 
 }
@@ -235,6 +236,8 @@ type Server struct {
 	gaKey        string
 	userVoiceKey string
 
+	TLSListenerOnly bool
+
 	ListenerString        string
 	TLSListenerString     string
 	ProfileListenerString string
@@ -261,7 +264,11 @@ func init() {
 }
 
 func (s *Server) Run() {
+	listening := false
+
 	if s.profilerEnabled {
+		listening = true
+
 		go func() {
 			fmt.Println("Profiled listening at: :6060")
 
@@ -363,21 +370,28 @@ func (s *Server) Run() {
 
 	mime.AddExtensionType(".md", "text/x-markdown")
 
-	log.Printf("Transfer.sh server started.\nlistening on port: %v\nusing temp folder: %s\nusing storage provider: %s", s.ListenerString, s.tempPath, s.storage.Type())
-	log.Printf("---------------------------")
+	log.Printf("Transfer.sh server started.\nusing temp folder: %s\nusing storage provider: %s", s.tempPath, s.storage.Type())
 
 	h := handlers.PanicHandler(handlers.LogHandler(LoveHandler(s.RedirectHandler(r)), handlers.NewLogOptions(log.Printf, "_default_")), nil)
 
-	srvr := &http.Server{
-		Addr:    s.ListenerString,
-		Handler: h,
+	if !s.TLSListenerOnly {
+		srvr := &http.Server{
+			Addr:    s.ListenerString,
+			Handler: h,
+		}
+
+		listening = true
+		log.Printf("listening on port: %v\n", s.ListenerString)
+
+		go func() {
+			srvr.ListenAndServe()
+		}()
 	}
 
-	go func() {
-		srvr.ListenAndServe()
-	}()
-
 	if s.TLSListenerString != "" {
+		listening = true
+		log.Printf("listening on port: %v\n", s.TLSListenerString)
+
 		go func() {
 			s := &http.Server{
 				Addr:      s.TLSListenerString,
@@ -391,48 +405,17 @@ func (s *Server) Run() {
 		}()
 	}
 
-	/*
-		cacheDir := "/var/cache/autocert"
-
-		if s.LetsEncryptCache != "" {
-			cacheDir = s.LetsEncryptCache
-		}
-
-		m := autocert.Manager{
-			Prompt: autocert.AcceptTOS,
-			Cache:  autocert.DirCache(cacheDir),
-			HostPolicy: func(_ context.Context, host string) error {
-				if !strings.HasSuffix(host, "transfer.sh") {
-					return errors.New("acme/autocert: host not configured")
-				}
-				return nil
-			},
-		}
-
-		if s.TLSListenerString != "" {
-			go func() {
-				s := &http.Server{
-					Addr:      ":https",
-					Handler:   lh,
-					TLSConfig: &tls.Config{GetCertificate: m.GetCertificate},
-				}
-
-				if err := s.ListenAndServeTLS("", ""); err != nil {
-					panic(err)
-				}
-			}()
-
-			if err := http.ListenAndServe(c.ListenerString, RedirectHandler()); err != nil {
-				panic(err)
-			}
-		}
-	*/
+	log.Printf("---------------------------")
 
 	term := make(chan os.Signal, 1)
 	signal.Notify(term, os.Interrupt)
 	signal.Notify(term, syscall.SIGTERM)
 
-	<-term
+	if listening {
+		<-term
+	} else {
+		log.Printf("No listener active.")
+	}
 
 	log.Printf("Server stopped.")
 }

